### ISSUE TITLE
[apex] AvoidDirectAccessTriggerMap incorrectly detects array access in classes

### ISSUE THREAD

**Issue by up2go-rsoesemann** [pmd/pmd#792]
**Affects Version:** 6.0.0
**Rule Set:**
ErrorProne > AvoidDirectAccessTriggerMap

**Description:**
Rule should only apply to `*.trigger` classes but detect arbitrary `*.cls` files with array [] brackets.

**Code Sample demonstrating the issue:**

```
public class NotATrigger {
   void foo(){
        Account[] accounts = opportunity.Accounts[0];
   }
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
all tools

**Comment by up2go-rsoesemann** [comment]
@JAertgeerts you originally contributed this rule. Can you please fix this?
